File: Front End/src/app/model/message.ts
import { User } from "./user";

export class Message{
    public id : string;
    public contenu_msg : string;
    public expediteur : User;
    public destinataire : User;
    public dateEnvoi : Date;
    public envoye : boolean;
    public readen: boolean;

    constructor(){
        this.id = '';
        this.contenu_msg = '';
        this.expediteur=null;
        this.destinataire = null;
        this.dateEnvoi = null;
        this.envoye = null;
        this.readen = null;
    }


}

File: ToSuceed/src/main/java/fr/ToSucceed/message/model/Message.java
package fr.ToSucceed.message.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import fr.ToSucceed.user.domain.User;
import lombok.*;

import javax.persistence.*;

import java.time.LocalDate;
import java.util.Date;

import static javax.persistence.GenerationType.AUTO;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Message {

    @Id
    @GeneratedValue(strategy = AUTO)
    private Integer id;
    private String contenu_msg;
    @ManyToOne
    @JoinColumn(name = "exp_id")
    @JsonIgnoreProperties("messages")
    private User expediteur;
    @ManyToOne
    @JoinColumn(name = "dest_id")
    @JsonIgnoreProperties("messages")
    private User destinataire;
    private String dateEnvoi;
    private boolean readen;





}
